packages/falso/src/lib/core/core.ts

```diff
@@ -22,7 +23,14 @@ export function fake<T, Options extends FakeOptions>(
   options?: Options
 ): Return<T, Options> {
   const dataSource = Array.isArray(data)
-    ? () => randElement(data)
+    ? () =>
+        randElement(
+          data.filter((fakeData) =>
+            options?.maxCharCount && typeof fakeData === 'string'
```

Comment:
@NetanelBasal Does it make sense that we have arrays with mixed data types? otherwise, I would test the array element and decide whether we need to filter.

We might want to add some more restrictions in the future, WDYT about creating an abstraction that applies these restrictions on the data set.

Reply:
@NetanelBasal any update?

Reply:
@NetanelBasal up

Reply:
Sorry for the late response. 

>  Does it make sense that we have arrays with mixed data types?

No

Reply:
@shaharkazaz what is your suggestion based on the response from Netanel?

Reply:
@shaharkazaz uo

Reply:
Given that we only have one data type per array we can move `options?.maxCharCount && typeof fakeData === 'string'` to only be tested once and not inside the filter loop.

Reply:
I am not sure I understand it properly. 

Do you suggest to code it like that

```
        randElement(
          options?.maxCharCount && typeof data[0] === 'string' ?
            data.filter(fakeData => fakeData.length <= options.maxCharCount!)
            : data
        )
```

Reply:
```ts
  const dataSource = (data as FactoryFunction<T>);
  if (Array.isArray(data)) {
   let resolvedData = data;
   if (options?.maxCharCount && isString(data[0])) {
     resolvedData = data.filter((item) => item.length <= options.maxCharCount);
   }
   dataSource = () => randElement(resolvedData);
  }
```

Reply:
@shaharkazaz done